Headings: accessibility, hierarchy, and cosmetics of in-line `code` therein

StevenBlack opened the pull request:
Two changes in the javascript button documentation, methods section: structural and cosmetic.

**Before** we had a `h2` - `h4` hierarchy.  The `h4` was javascript code.

Changing to a `h2` - `h3` hierarchy looks kind of wonky since the `h3` is code, and it heads a single-line explanation.

**After** — in this pull request — we have an `h2` with a definition list.

Note I added a `p` tag on the `dd` elements to get a better looking bottom margin between `dt` elements in the list.

![bs-button](https://cloud.githubusercontent.com/assets/80144/6363526/61115212-bc68-11e4-8f42-d96b3a8bc353.jpg)

StevenBlack committed:
Accessibility and cosmetic changes to headings

StevenBlack committed:
Accessibility and cosmetic changes to headings

mdo commented:
We'd probably want the same styles on all our code headings then, no? We have those in a lot of places.

StevenBlack commented:
I count 29 instances, yes.   If you prefer `code` in definition lists rather than headings (my guess: yes) then I'm on it.

` ack "h4.*\("`

    alerts.html
    39:  <h4>$().alert()</h4>
    42:  <h4>$().alert('close')</h4>
    
    carousel.html
    198:  <h4>.carousel(options)</h4>
    206:  <h4>.carousel('cycle')</h4>
    209:  <h4>.carousel('pause')</h4>
    213:  <h4>.carousel(number)</h4>
    216:  <h4>.carousel('prev')</h4>
    219:  <h4>.carousel('next')</h4>
    
    collapse.html
    223:  <h4>.collapse(options)</h4>
    231:  <h4>.collapse('toggle')</h4>
    234:  <h4>.collapse('show')</h4>
    237:  <h4>.collapse('hide')</h4>
    
    dropdowns.html
    162:  <h4>$().dropdown('toggle')</h4>
    
    modal.html
    459:  <h4>.modal(options)</h4>
    467:  <h4>.modal('toggle')</h4>
    471:  <h4>.modal('show')</h4>
    475:  <h4>.modal('hide')</h4>
    
    popovers.html
    251:  <h4>$().popover(options)</h4>
    254:  <h4>.popover('show')</h4>
    258:  <h4>.popover('hide')</h4>
    262:  <h4>.popover('toggle')</h4>
    266:  <h4>.popover('destroy')</h4>
    
    scrollspy.html
    97:  <h4>.scrollspy('refresh')</h4>
    
    tabs.html
    95:  <h4>$().tab</h4>
    
    tooltips.html
    212:  <h4>$().tooltip(options)</h4>
    215:  <h4>.tooltip('show')</h4>
    219:  <h4>.tooltip('hide')</h4>
    223:  <h4>.tooltip('toggle')</h4>
    227:  <h4>.tooltip('destroy')</h4>

StevenBlack committed:
Added code tags to method sub-headings

StevenBlack committed:
Added code tags to method sub-headings

StevenBlack commented:
OK I hit all 29 instances. The general case has more text than `buttons.html` so I reverted `buttons.html` back to using `h4` tags.  Starting now all `h4` tags have `code` tags where appropriate.

Here's what it looks like now, using `modal.html` here as an example.

![bs-modal](https://cloud.githubusercontent.com/assets/80144/6366127/13d58a64-bc8f-11e4-853d-b5658c1dbf6e.jpg)

patrickhlauke commented:
LGTM

hnrch02 commented:
@mdo You down with this?

StevenBlack committed:
Code in headings retain heading background color

StevenBlack committed:
Code in headings retain heading background color

StevenBlack commented:
Slightly larger for you.  I think this looks really good.

![codeinh4-2](https://cloud.githubusercontent.com/assets/80144/6418529/bfa872ac-be83-11e4-9c29-be8cd5e506d1.jpg)

StevenBlack committed:
Covering all instances of in-line code in headings

As a result of #15946 there are other instances in documentation where we have in-line code in headings.  Fixed.

StevenBlack committed:
Covering all instances of in-line code in headings

As a result of #15946 there are other instances in documentation where we have in-line code in headings.  Fixed.

StevenBlack commented:
The CSS commit above covers some instances of code-in-headings not seen in the components section of the docs.

Here's a before-and-after look at code in larger headings, code that's actually in-line with some text.

![bs-code-headings](https://cloud.githubusercontent.com/assets/80144/6419481/8de920c4-be8b-11e4-8638-aed5a369fbfd.jpg)

patrickhlauke commented:
LGTM as well. Still waiting for @mdo's blessing :)

XhmikosR commented:
:+1: from me too

cvrebert commented:
I am very much in favor of using `<code>` for method headings in the API docs like this.

XhmikosR commented:
@StevenBlack: can you fetch and rebase?

XhmikosR commented on docs/assets/css/src/docs.css:
```diff
@@ -28,6 +28,11 @@ body {
   font-weight: normal;
 }
 
+/* In-line code within headings retain the heading background */
```
It's `inline` and not `in-line` as far as I can tell.

Also, break one selector per line.

StevenBlack replied:
@XhmikosR  Confirming: `in-line` is correct usage.

XhmikosR replied:
I'd rather have it `inline`.

XhmikosR commented:
@StevenBlack: Also squash the patches and take care of my previous comment please.

patrickhlauke committed:
Merge pull request #15974 from patrickhlauke/ios-click-fix-role-button2

Add upstream patch comment for #15947

twbs-grunt committed:
automatic grunt dist

StevenBlack committed:
"inline", and one line per selector.

StevenBlack committed:
Merge branch 'docs.js.button' of https://github.com/StevenBlack/bootstrap into docs.js.button

* 'docs.js.button' of https://github.com/StevenBlack/bootstrap:
  Covering all instances of in-line code in headings
  Code in headings retain heading background color
  Added code tags to method sub-headings
  Accessibility and cosmetic changes to headings

Conflicts:
	docs/_includes/js/popovers.html
	docs/_includes/js/tooltips.html
	docs/assets/css/src/docs.css

StevenBlack committed:
Manual conflict resolution.

StevenBlack commented:
Having no success squashing.

kkirsche commented:
@StevenBlack `git rebase -i HEAD~13 [your-branch-name`

You'll be presented with a list of commits that say `pick [something] "commit message"`.

Change all except the first `pick` with `squash`

cvrebert commented:
Looks like the rebase had some problems. I'm happy to do the squashing when merging this. Just waiting on @mdo to approve the style changes.

XhmikosR commented:
Don't specify the number of commits. Just run git rebase --i
upstream/master after you have fetched upstream.
On Mar 2, 2015 10:43 PM, "Kevin Kirsche" <notifications@github.com> wrote:

> @StevenBlack <https://github.com/StevenBlack> git rebase -i HEAD~13
> [your-branch-name
>
> You'll be presented with a list of commits that say pick [something]
> "commit message".
>
> Change all except the first pick with squash
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twbs/bootstrap/pull/15915#issuecomment-76815346>.
>

StevenBlack commented:
And we do this busy-work because....?   Just curious.

XhmikosR commented:
Because it's useless to have many patches for the same thing apart from
making git history bigger. Also, it's sometimes mandatory to rebase anyway
due to conflicts.
On Mar 2, 2015 10:55 PM, "Steven Black" <notifications@github.com> wrote:

> And we do this busy-work because....? Just curious.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twbs/bootstrap/pull/15915#issuecomment-76819478>.
>

kkirsche commented:
Ah. Sorry @XhmikosR I've had bad luck doing it that way. As long as it works though in the end that's what matters.

But yeah, @StevenBlack having a single commit to point to for a patch, in my opinion, can make it easier to identify where a regression or bug was introduced and trace it back to the correct issue so you don't have 15+ commits for a single patch.

Just my opinion though :smiley:
